test/unix/main.ml

```diff
@@ -143,23 +238,86 @@ let readonly () =
               (Printf.sprintf "Wrong insertion: %s value is missing." v))
     !l
 
+let readonly_no_flush () =
```

Comment:
Actually I think this is not part of the requirements that this should pass.
With the current implementation it's not passing for every situation (try playing with `index_size` for instance.
Sticking to the current requirement, ie reading after a flush must be correct is enough for now.